lib/api/BackbeatAPI.js

```diff
@@ -615,6 +622,150 @@ class BackbeatAPI {
         err => cb(err, response));
     }
 
+    /**
+     * Tempfix to correct failure/pending metrics on concurrency issue
+     * @param {array} keys - an array of redis failure keys to remove.
+     *   Schema:
+     *   bb:crr:failed:<bucket>:<key>:<versionID>:<storageClass>:<timestamp>
+     * @return {undefined}
+     */
+    _normalizeFailMetrics(keys) {
+        // ignore deleting duplicate keys for tests
+        if (process.env.CI === 'true') {
+            return;
+        }
+        async.eachLimit(keys, 10, (key, next) => {
+            async.waterfall([
+                done => this._redisClient.batch([['get', key]], (err, res) => {
+                    if (err) {
+                        // error occurred, skip for next time
+                        return done(err);
+                    }
+                    const [cmdErr, sourceRole] = res[0];
+                    if (cmdErr) {
+                        // error occurred, skip for next time
+                        return done(cmdErr);
+                    }
+                    if (!sourceRole) {
```

Comment:
I don't get the link between `sourceRole` missing and the key being deleted. If this is correct, can you comment to explain why?

Reply:
When making the `get` to redis, if no key was found in redis, `sourceRole` will be null

Reply:
Yes, but why `sourceRole`? Is it the value stored in the key?

Reply:
Or could be the argument needed to create the `ObjectFailureEntry`

Reply:
yes, `sourceRole` is the arn saved in the key